doc/e2e-tests.md

```diff
@@ -0,0 +1,215 @@
+# End to end test documentation
+
+Running e2e ensures various parts of the Compliance Operator are working correctly in a real cluster environment. And we are able to test out new rules and new features using e2e test.
+The tests file directories for e2e are under tests/e2e/
+
+`tests/e2e/e2e-tests.md` - contains all the tests we have, we can add new tests to that file.
+
+## How to add a new test
+
+First, the structure for the test is following:
+
+```go
+	testExecution{
+			Name:       "TestKubeletConfigRemediation",
+			IsParallel: false,
+			TestFn: func(t *testing.T, f *framework.Framework, ctx *framework.Context, mcTctx *mcTestCtx, namespace string) error {
+            ...
+            }
+    }
+```
+We can set `IsParallel` to instruct the test run as Parallel or Serial.
+
+### Build Content image and create `ProfileBundle`
```

Comment:
I'm fine having this documented for now, but I think we should have a Jira task to get rid of using quay images for tests (and I know that I started that dubious practice..). We should move to having all the test content in the repo and build the container images during the test..

This is not criticizing your document, just saying we should at least plan on fixing this part of our e2e testing..

Reply:
Thanks for the review, I will create a Jira task for that!